packages/webpack-cli/lib/utils/Compiler.js

```diff
@@ -80,6 +80,9 @@ class Compiler {
             logger.error(err.stack || err);
             process.exit(1); // eslint-disable-line
         }
+        if (!outputOptions.watch && stats.hasErrors()) {
+            process.exitCode = 1;
+        }
```

Comment:
warnings and errors with `1`, here only errors, we need to test on warnings too, you can create simple loader and use `this.emitWarning` to test it